domain/stub/service.go

```diff
@@ -37,26 +42,62 @@ type StubService struct {
 	modelState      *domain.StateBase
 	controllerState *domain.StateBase
 
-	providerWithSecretToken providertracker.ProviderGetter[ProviderWithSecretToken]
+	providerWithSecretToken      providertracker.ProviderGetter[ProviderWithSecretToken]
+	providerForAgentBinaryFinder providertracker.ProviderGetter[ProviderForAgentBinaryFinder]
+
+	getPreferredSimpleStreams PreferredSimpleStreamsFunc
+	agentBinaryFilter         AgentBinaryFilter
 }
 
+// PreferredSimpleStreamsFunc is a function that returns the preferred streams
+// for the given version and stream.
+type PreferredSimpleStreamsFunc func(
+	vers *semversion.Number,
+	forceDevel bool,
+	stream string,
+) []string
+
+// AgentBinaryFilter is a function that filters agent binaries based on the
+// given parameters. It returns a list of agent binaries that match the filter
+// criteria.
+type AgentBinaryFilter func(
```

Comment:
Should this be `AgentBinaryFinder`? Filtering is an internal implementation mechanism it seems like.